src/data/finsupp/convolution.lean

```diff
@@ -0,0 +1,270 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
```

Comment:
Hmmm, I'm not sure if I like the filename. (Also you could add your name to the list of authors, if you want.)
I think the filename should contain `monoid_algebra`, and probably we can move this out of the `finsupp` directory, into `data/` itself.
That's where people might look for this.

Reply:
Okay, I've moved this back to `data/monoid_algebra`, and added my name.